packages/@aws-cdk/aws-ecs/lib/base/base-service.ts

```diff
@@ -62,6 +62,21 @@ export interface BaseServiceProps {
    * Options for enabling AWS Cloud Map service discovery for the service
    */
   readonly serviceDiscoveryOptions?: ServiceDiscoveryOptions;
+
+  /**
+   * Whether the new long ARN format has been enabled on ECS services.
+   * NOTE: This assumes customer has opted into the new format for the IAM role used for the service, and is a
+   * workaround for a current bug in Cloudformation in which the service name is not correctly returned when long ARN is
+   * enabled.
+   *
+   * Old ARN format: arn:aws:ecs:region:aws_account_id:service/service-name
+   * New ARN format: arn:aws:ecs:region:aws_account_id:service/cluster-name/service-name
+   *
+   * See: https://docs.aws.amazon.com/AmazonECS/latest/userguide/ecs-resource-ids.html
```

Comment:
Could you please also create a GitHub issue to track the upstream bug and this workaround and refer to it here? Or maybe use the issue that we already have? I'd like a record of issues in CloudFormation we're working around so we can take out the workarounds once they fix it upstream.

Reply:
https://github.com/awslabs/aws-cdk/issues/2184